### ISSUE TITLE
Not able to load data on Kobo Clara 2e

### ISSUE THREAD

**Issue by meranders** [athombv/homey.ink#36]
Tried this on my Kobo Clara 2e, but it will not load any data. Just says n/a. The token url works fine on a desktop machine. Could there be some changes to the Clara 2e browser vs the previous ones, maybe Javascript is disabled?

**Comment by tr1pp4** [comment]
Would also be excited to see support for other e-readers from Kobo, since the Aura line seems to be no longer on the market.  Clara 2e would be a good choice

**Comment by jonasmann** [comment]
I have the same issue, but on a Kobo Aura H2O. N/A on all data and it does not rotate.
But I also had a lot of issue with saving the file on the kobo. It was like it wouldn't recognise the transfer and only kept the old version of the .conf file.

**Comment by dsellig** [comment]
> I tried this on my Kobo Clara 2e, but it will not load any data. It just says n/a. The token URL works fine on a desktop machine. Could there be changes to the Kobo Clara 2e web browser settings vs. the previous ones? Maybe Javascript is disabled?

Ever managed to get this working?

**Comment by malnor** [comment]
Yeah, seems like a javascript problem. :-/

**Comment by cas541** [comment]
Is there already a fix for the problem above? 
I have tried it last week om my Aura h2o, but I have the same problem?

**Comment by oskarirauta** [comment]
@malnor @cas541 @meranders @WeeJeWel @Dijker 

Any solution? I just bought Aura HD for this and now it's useless. Alternative HomeyDash kind of works.. But looks terrible on monochrome display. It's not just a javascript problem- try url with computer's browser to end up with same result. Replace beginning with homeydash.com/?theme=web&token=.... and it works fine, but it isn't homey.ink....

So there's a chance that it's javascript related, but not what you thought.. Javascript console lists few errors, where the most critical is "XMLHttpRequest cannot load https://10-0-0-194.homey.homeylocal.com/api/manager/system/ping?id=65a7c357223e95eedc25b7bf due to access control checks."

10.0.0.194 is my Homey's ip address in local network. I tried to access this url with other tools (curl) and it's a miss. I searched for details on this address, and X-X-X-X.homey.homeylocal.com is supposed to be valid https redirection to local http service of homey ackording to Homey API documentation, but api/manager/system/ping fails... Seems like URI doesn't exist, but hard to say without chances to see logs of homey's web server, or config and published files on it.

@jonasmann 
Problem probably was that you had tried solitaire on it.. It breaks everything. To fix that problem, copy the original/working configuration to computer and edit the file, remove anything on General section. After this is it should work. Playing games stores something (probably card decks card order) which is parsed by homey.ink incorrectly and result is corrupted configuration.

It's amazing that there isn't more people making this a issue, images of homey.ink just look amazing and I would think this would much more popular; maybe homeydash.com on iPad has bells and whistles, but sometimes less is more. Kobo screen doesn't work as a flash light and bug anyone while watching tv or something, and has other huge advantages as well. Ok, disadvantages as well, but your mileage varies, so list of pros and cons isn't static.

**Comment by WeeJeWel** [comment]
I might've fixed it. It was broken on my H2O, and after some changes it works again.

It now also shows an error when it doesn't load.

**Comment by oskarirauta** [comment]
Thanks, I'll test it first thing tomorrow! :)

**Comment by oskarirauta** [comment]
Okay, still work in progress..

https://10-0-0-194.homey.homeylocal.com/api/manager/system/ping?id=XXXXX

this url doesn't lead to anywhere. I replace protocol and domain with http://10.0.0.194 (in computer browser window) and it resulted as true.

**Comment by oskarirauta** [comment]
I see that this pinging isn't a problem, homeydash has same issue. Maybe it's related to my home's firewall..
For now, page displays date and weather correctly;  but favourite devices and flows is empty. This then isn't about my firewall, as they work with homeydash..

**Comment by oskarirauta** [comment]
Okay, here's some; all errors fixed (sorry, load event calls now function fillData, instead of running function inline; I had to call the function manually at some point, that's redundant now..)

 - Lot of code from homeydash providing support for devices not previously supported.
 - Lot of modifications to that code and to already existing code
 - changes to css
 - flows untested
 - devices tested with single temperature sensor, sorry
 - displayed property of device can be cycled if multiple measurement values are available by touching the device icon
 - no clue how a switch device will perform.. We should have on/off status display for that.. And maybe there already is?

More details on commit history (open commits to see details).

https://github.com/oskarirauta/homey.ink

**Comment by WeeJeWel** [comment]
Hm, I've just tested, and on my Kobo H2O I can see the Favorite Flows & Favorite Devices.

**Comment by oskarirauta** [comment]
Yes; but I have a temperature sensor which provides following: battery state, battery voltage, temperature - and it doesn't display any of following details on my HD, touching the device should cycle values, but it doesn't do anything; it does work in my computer's browser though.
